File: V5_demo/wss_demo/c#/ws_public_demo.cs
using System.Net.WebSockets;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;


string topic = "orderbook.50.BTCUSDT";

await ConnectWS();

async Task ConnectWS()
{
    using (var ws = new ClientWebSocket())
    {
        await ws.ConnectAsync(new Uri("wss://stream-testnet.bybit.com/v5/public/linear"), CancellationToken.None);
        await OnOpen(ws);
        await Receive(ws);
    }
}

async Task OnOpen(ClientWebSocket ws)
{
    Console.WriteLine("opened");
    var subscribeRequest = JsonConvert.SerializeObject(new
    {
        op = "subscribe",
        args = new string[] { topic }
    });
    await ws.SendAsync(Encoding.UTF8.GetBytes(subscribeRequest), WebSocketMessageType.Text, true, CancellationToken.None);
}

async Task Receive(ClientWebSocket ws)
{
    var buffer = new ArraySegment<byte>(new byte[8192]);
    while (true)
    {
        var result = await ws.ReceiveAsync(buffer, CancellationToken.None);
        if (result.MessageType == WebSocketMessageType.Close)
        {
            Console.WriteLine("### about to close ###");
            break;
        }
        else if (result.MessageType == WebSocketMessageType.Text)
        {
            if (buffer.Array != null)
            {
                var message = Encoding.UTF8.GetString(buffer.Array, 0, result.Count);
                var data = JsonConvert.DeserializeObject<JObject>(message);
                if (data != null)
                {
                    Console.WriteLine(data.ToString());
                }
            }
        }
    }
}

File: V5_demo/wss_demo/go/ws_public_demo.go
package main

import (
	"encoding/json"
	"fmt"
	"github.com/gorilla/websocket"
	"log"
	"time"
)

func main() {
	topic := "orderbook.50.BTCUSDT"
	connectWS(topic)
}

func connectWS(topic string) {
	address := "wss://stream-testnet.bybit.com/v5/public/linear"
	c, _, err := websocket.DefaultDialer.Dial(address, nil)
	if err != nil {
		log.Fatal("Failed to connect:", err)
	}
	defer c.Close()

	fmt.Println("Connected.")

	subscribe(c, topic)

	for {
		_, message, err := c.ReadMessage()
		if err != nil {
			log.Println("Failed to read message:", err)
			return
		}
		receive(string(message))
	}
}

func subscribe(c *websocket.Conn, topic string) {
	subscribeRequest := map[string]interface{}{
		"op":   "subscribe",
		"args": []string{topic},
	}
	message, err := json.Marshal(subscribeRequest)
	if err != nil {
		log.Println("Failed to marshal JSON:", err)
		return
	}
	err = c.WriteMessage(websocket.TextMessage, message)
	if err != nil {
		log.Println("Failed to send subscribe message:", err)
		return
	}
}

func receive(message string) {
	var data map[string]interface{}
	err := json.Unmarshal([]byte(message), &data)
	if err != nil {
		log.Println("Failed to unmarshal JSON:", err)
		return
	}
	fmt.Println(data)
}

func keepAlive(c *websocket.Conn) {
	for {
		err := c.WriteMessage(websocket.PingMessage, nil)
		if err != nil {
			log.Println("Ping failed:", err)
			return
		}
		time.Sleep(time.Second * 10)
	}
}
